tests/unit/test-firefox/test.firefox.js

```diff
@@ -545,6 +646,136 @@ describe('firefox', () => {
 
   });
 
+  describe('defaultCreateProfileFinder', () => {
+
+    it('creates a finder', async () => {
+      const FxProfile = {
+        Finder: sinon.spy(() => () => Promise.resolve()),
+      };
+      firefox.defaultCreateProfileFinder({FxProfile});
+      sinon.assert.called(FxProfile.Finder);
+      sinon.assert.calledWith(
+        FxProfile.Finder,
+        sinon.notOk,
+      );
+    });
+
+    it('creates finder based on userDirectoryPath if present', async () => {
+      const FxProfile = {
+        Finder: sinon.spy(() => () => Promise.resolve()),
+      };
+      const userDirectoryPath = '/non/existent/path';
+
+      firefox.defaultCreateProfileFinder({userDirectoryPath, FxProfile});
+
+      sinon.assert.called(FxProfile.Finder);
+      sinon.assert.calledWith(
+        FxProfile.Finder,
+        sinon.match(userDirectoryPath),
+      );
+    });
+
+    it('returns a finder that looks for a default profile', async () => {
+      const FxProfile = {
```

Comment:
If I'm reading this test correctly, we could make this a bit more readable by doing the opposite:

```js
         const fakeReadProfiles = sinon.spy(() => {
           return Promise.reject(
             new Error('unspecified error'));
         });
         const fakeGetPath = sinon.spy(() => Promise.resolve());
         const fakeProfiles = [{Name: 'someName'}];

         const FxProfile = {
           Finder: function () {
             return {
               readProfiles: fakeReadProfiles,
               getPath: fakeGetPath,
               profiles: fakeProfiles,
             };
          },
         };

         ....
```

This is a pattern used in a couple of tests, and so the same comment is valid for the tests below, it may be even worth to turn it into a test helper function defined in this test file and reuse it in all the tests.